### PR TITLE
Remove header for LG TV apps

### LINKED ISSUE #125: Jellyfin for webOS client and nginx X-Frame-Options
### Describe the bug

The Jellyfin for webOS client cannot connect to a Yunohost-installed Jellyfin server

### Context

- Hardware: RPI4
- YunoHost version: 11.0.11
- I have access to my server:  ssh and web admin
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: no
- Using, or trying to install package version/branch: master

### Steps to reproduce

Install the new official client "Jellyfin for WebOS" listed [here](https://jellyfin.org/downloads/clients/)
Enter the information of your server.
An error appears (error 27).

### Expected behavior

No error on connection

### Logs

N/A

### Additional comments

This is a documented error - if you look at the documentation of jellyfin [here](https://jellyfin.org/docs/general/networking/nginx/), there is a tip mentioning that 

> The default X-Frame-Options header may cause issues with the webOS app, causing it to remain stuck at a black screen. If enabled, the default Content Security Policy may also cause issues.

If you look at the `/etc/nginx/conf.d/security.conf.inc` file of Yunohost, on [this line](https://github.com/YunoHost/yunohost/blob/ce40408f1b1c47cc59d44a6859f9e224c6104294/conf/nginx/security.conf.inc#L37), you see that the X-Frame-Options header is indeed set to SAMEORIGIN.

I have tested a dirty fix by commenting this line, restarting nginx, and the WebOS client manages to connect.

But, it's a *very* dirty fix. I also tried to add `add_header X-Frame-Options "";` to `/etc/nginx/conf.d/my.domain.d/jellyfin.conf` but it didn't solve the problem.

### PR DESCRIPTION
**Pull request by noahm**
## Problem

Fixes #125

## Solution

- Updated nginx config to exclude the `X-Frame-Options` header when responding to clients sending `WebOS` in their user agent.
- I have manually tested and confirmed the fix with my own TV.

## PR Status

- [x] Code finished and ready to be reviewed/tested
- [x] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by tituspijean** [comment]
I actually now experience this issue too, thanks for this PR! I will try it out by Thursday. 🤞

**Comment by tituspijean** [comment]
Using `if` in NGINX is generally discouraged due to poorer performance, si it got me looking into the issue. Could you have a look to this forum thread? https://forum.jellyfin.org/t-solved-webos-error-code-27

It looks like specifying without the `if` block `more_set_headers X-Frame-Options : "ALLOW-FROM https://__DOMAIN____PATH__";` would work? Could you try it?

**Comment by noahm** [comment]
Ah, interesting. I'll try that and report back.

**Comment by noahm** [comment]
> Using `if` in NGINX is generally discouraged due to poorer performance

Do you have any documentation for this? I did some searches and couldn't come up with anything.
 
> It looks like specifying without the `if` block `more_set_headers X-Frame-Options : "ALLOW-FROM https://__DOMAIN____PATH__";` would work? Could you try it?

I looked into this a bit further and[ the `ALLOW-FROM` directive of X-Frame-Origins is obsolete](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options#allow-from_origin) and not respected by any browser shipped in many years. This means it's essentially equivalent to simply omitting the X-Frame-Options header entirely.

The successor and modern alternative is the `Content-Security-Policy` header and its `frame-ancestors` directive. I tested with several different variants of that and all of them broke the LG app, but in a different way than before. With the current `X-Frame-Options` present, an "Unable To Load" error screen is presented with error code 27. With the CSP `frame-ancestors` set instead, the app appears to boot, but it gets stuck on a blank dark grey screen without any content or error message at all.

In summary I think the only two functioning, reasonable options are either:

1. Simply remove the `X-Frame-Options` header from jellyfin responses unconditionally, accepting this allows the possibility someone can iframe-embed your jellyfin server into other places on the web.
2. Attempt to preserve existing security posture at the cost of some performance by remove the header conditionally with the if directive I started with

**Comment by tituspijean** [comment]
So, it was actually a bad remembrance on my part, the old NGINX wiki mentions that "if is evil", but not due to performance: https://github.com/nginxinc/nginx-wiki/blob/836ecd605a1b9861fb608e848336bca9b8640b54/source/start/topics/depth/ifisevil.rst

Either we remove the header with `more_set_headers X-Frame-Options : "";` or we implement what you suggested in this PR. I'll ping other packagers, I actually have no idea what's best or worst.

**Comment by noahm** [comment]
Wow, that's a fascinating hole to dive down. I read through everything and the other explanation blog post linked at the end and I think the use case I've proposed will work as long as a second if directive isn't added in the future?

I'm also curious, is this advice still relevant? The page seems to have been taken down and there's no mention of the dangers in the official docs on the if statement.

**Comment by tituspijean** [comment]
Indeed, it has been removed, and the performance impact I imagined should be minimal. As for safety, let's minimize whose clients we remove the header from, so let's go with your code. :)
